### ISSUE TITLE
Clarify `segment_folders()` fn behavior

### ISSUE THREAD

**Issue by hermancollin** [axondeepseg/axondeepseg#813]
I'm opening an issue for this because figuring this out confused me a little bit. For context, the _segment.py_ file contains 2 major functions: `segment_image()` and `segment_folders()`. The former was used for single image inference whereas the latter was used... for what exactly? This is the `segment_folders` signature on the master branch:

```python
def segment_folders(
    path_testing_images_folder, 
    path_model,
    overlap_value, 
    acquired_resolution=None,
    zoom_factor=1.0,
    no_patch=False,
     gpu_id=0,
     verbosity_level=0
):
```

- the plural form in the name of the function implies that we are segmenting multiple folders
- the `path_testing_images_folder` argument is a single path (not a list). This is how its documented: 
> the folder where all image folders are located (the images to segment are located in those image folders)
- this implies that this function segments multiple folders nested inside the specified directory 
- HOWEVER, when we look at the actual implementation (see [these lines](https://github.com/axondeepseg/axondeepseg/blob/0a5cf545fedd9cb6cb7137aeaff040e863541904/AxonDeepSeg/segment.py#L239-L240)), `pathlib.Path.iterdir()` is called on `path_testing_images_folder` and images are selected directly inside this directory.

In other words, the docstring and the implementation do not match. The current behavior of this function is to segment all images inside the given folder, which makes sense really. A correct name for this function would be `segment_folder()` (no plural).

Related to #800 and #798

(I suspect this messy code is the result of many refactoring.)

**Comment by mathieuboudreau** [comment]
This makes sense to me - OK with renaming the function (or tbh, even merging the two would probably be fine)

**Comment by mathieuboudreau** [comment]
Is this going to get done in the #800 PR, or should we tackle it in a different branch?

**Comment by hermancollin** [comment]
@mathieuboudreau yes, this is all included in #800.
